chalicelib/app_utils.py

```diff
@@ -209,7 +212,7 @@ def process_response(response):
     Does any final processing of a Foursight response before returning it. Right now, this includes:
     * Changing the response body if it is greater than 5.5 MB (Lambda body max is 6 MB)
     """
-    if len(json.dumps(response.body)) > 5500000:
+    if get_size(response.body) > 5500000:  # should be much faster than json.dumps
```

Comment:
I'm not sure if get_size is a builtin of some sort or something we wrote. If a builtin, I might still write a separate function (untested):
```
def response_body_size_exceeds(response, max_size):
    return get_size(response.body) > max_size  # should be faster than measuring against json.dumps
```
so you can separate heuristics from policy. Then (untested):
```
HUGE_RESPONSE_BODY_THRESHOLD = 5500000

def process_response(response):
    if response_body_size_exceeds(response, HUGE_RESPONSE_BODY_THRESHOLD):
        response.body = ("Body size exceeded %s MB maximum."
                         % HUGE_RESPONSE_BODY_THRESHOLD / 1.0e6)
        response.status_code = 413
    return response
```
Although I might have introduced other bugs along the way, this would among other things make the function not lie about the size it's looking at, which is 5.5 MB, not 6 MB, though if you wanted to blur it you could still make the code less fragile by just doing the division and rounding up. The real point here is to insulate you against the message being wrong if the threshold number is changed and someone doesn't notice that the string is trying to be sync'd with that value.

This modularity also means you could choose to optimize the response body calculation to stop after it knew it was bigger than the threshold amount.